grid2op/multi_agent/multiAgentEnv.py

```diff
@@ -0,0 +1,557 @@
+                                                     cent_env_cls.load_to_subid
+        )
+        tmp_subgrid.gen_to_subid  = self._relabel_subid(tmp_subgrid.mask_gen,
+                                                     new_label,
+                                                     cent_env_cls.gen_to_subid
+        )
+        tmp_subgrid.storage_to_subid  = self._relabel_subid(tmp_subgrid.mask_storage,
+                                                        new_label,
+                                                        cent_env_cls.storage_to_subid
+        )
+        tmp_subgrid.line_or_to_subid = self._relabel_subid(tmp_subgrid.mask_line_or,
+                                                        new_label,
+                                                        cent_env_cls.line_or_to_subid
+        )
+        tmp_subgrid.line_ex_to_subid = self._relabel_subid(tmp_subgrid.mask_line_ex,
+                                                        new_label,
+                                                        cent_env_cls.line_ex_to_subid
+        )
+        
+        interco_to_subid = np.zeros(cent_env_cls.n_line, dtype=dt_int) - 1
+        tmp_ = np.zeros(tmp_subgrid.n_interco, dtype=dt_int) - 1
+        tmp_[tmp_subgrid.interco_is_origin] = cent_env_cls.line_or_to_subid[tmp_subgrid.mask_interco][tmp_subgrid.interco_is_origin]
+        tmp_[~tmp_subgrid.interco_is_origin] = cent_env_cls.line_ex_to_subid[tmp_subgrid.mask_interco][~tmp_subgrid.interco_is_origin]
+        interco_to_subid[tmp_subgrid.mask_interco] = tmp_
+        tmp_subgrid.interco_to_subid = self._relabel_subid(tmp_subgrid.mask_interco,
+                                                           new_label,
+                                                           interco_to_subid)
+        
+        tmp_subgrid.interco_to_lineid = np.arange(cent_env_cls.n_line)[tmp_subgrid.mask_interco]
+        tmp_subgrid.name_interco = np.array([
+            f'interco_{i}_line_{tmp_subgrid.interco_to_lineid[i]}' for i in range(len(tmp_subgrid.interco_is_origin))
+        ])
+
+        # which index has this element in the substation vector 
+        tmp_subgrid.load_to_sub_pos = cent_env_cls.load_to_sub_pos[tmp_subgrid.mask_load]
+        tmp_subgrid.gen_to_sub_pos = cent_env_cls.gen_to_sub_pos[tmp_subgrid.mask_gen]
+        tmp_subgrid.storage_to_sub_pos = cent_env_cls.storage_to_sub_pos[tmp_subgrid.mask_storage]
+        tmp_subgrid.line_or_to_sub_pos = cent_env_cls.line_or_to_sub_pos[tmp_subgrid.mask_line_or]
+        tmp_subgrid.line_ex_to_sub_pos = cent_env_cls.line_ex_to_sub_pos[tmp_subgrid.mask_line_ex]
+        
+        tmp_ = np.zeros(tmp_subgrid.n_interco, dtype=dt_int) - 1
+        tmp_[tmp_subgrid.interco_is_origin] = cent_env_cls.line_or_to_sub_pos[tmp_subgrid.mask_interco][tmp_subgrid.interco_is_origin]
+        tmp_[~tmp_subgrid.interco_is_origin] = cent_env_cls.line_ex_to_sub_pos[tmp_subgrid.mask_interco][~tmp_subgrid.interco_is_origin]
```

Comment:
Write a comment here to make sure you understood what's going on here :-)

Reply:
Done